js/ui/appActivation.js

```diff
@@ -139,8 +145,10 @@ const AppActivationContext = new Lang.Class({
     _maybeClearSplash: function() {
         // Clear the splash only when we've waited at least 700ms,
         // and when the app has transitioned to the running state...
-        if (this._appStateId == 0 && this._timeoutId == 0)
+        if (this._appStateId == 0 && this._timeoutId == 0) {
+            this._resetWindowsVisibility();
```

Comment:
I'd swear I moved it... Maybe it just copied it but I'm still surprised it got through. (I always use add -p for reviewing the changes I add)